src/main/java/Dude.java

```diff
@@ -0,0 +1,213 @@
+import java.io.*;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Dude {
+
+    static ArrayList<Task> taskList = new ArrayList<Task>();
+    static int nTasks = 0;
+
+    private static final String FILE_PATH = "./data/dude.txt";
+
+    public static void addTodo(String task) {
+        ToDo newTask = new ToDo(task);
+//        taskList[nTasks] = newTask;
+        taskList.add(newTask);
+        System.out.printf("Got it. I've added this task:\n%s\n", newTask.toString());
+        nTasks += 1;
+        System.out.printf("Now you have %d tasks in the list. \n", nTasks);
+    }
+    public static void addDeadline(String task, LocalDateTime by) {
+        Deadline newTask = new Deadline(task, by);
+//        taskList[nTasks] = newTask;
+        taskList.add(newTask);
+        System.out.printf("Got it. I've added this task:\n%s\n", newTask.toString());
+        nTasks += 1;
+        System.out.printf("Now you have %d tasks in the list. \n", nTasks);
+    }
+    public static void addEvent(String task, String from, String to) {
+        Event newTask = new Event(task, from, to);
+//        taskList[nTasks] = newTask;
+        taskList.add(newTask);
+        System.out.printf("Got it. I've added this task:\n%s\n", newTask.toString());
+        nTasks += 1;
+        System.out.printf("Now you have %d tasks in the list. \n", nTasks);
+    }
+
+    public static void delete(int n) {
+        Task removedTask = taskList.get(n-1);
+        taskList.remove(n-1);
+        nTasks -= 1;
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(removedTask.toString());
+        System.out.printf("Now you have %d tasks in the list.\n", nTasks);
+
+    }
+
+    public static void list() {
+        for (int i = 0; i < nTasks; i++) {
+            Task task = taskList.get(i);
+//            Task task = taskList[i];
+            System.out.printf("%d. %s\n", i+1, task.toString());
+        }
+    }
+
+    public static void mark(int n) {
+        taskList.get(n-1).setDone(true);
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.printf("%d. %s\n", n, taskList.get(n-1).toString());
+    }
+
+    public static void unmark(int n) throws IOException {
+        taskList.get(n-1).setDone(false);
```

Comment:
Operators should be surrounded by a space character.
```suggestion
        taskList.get(n - 1).setDone(false);
```